src/backends/torch/torchutils.cc

```diff
@@ -57,7 +57,12 @@ namespace dd
       if (!value.isTensor())
         throw MLLibInternalException("Expected Tensor, found "
                                      + value.tagKind());
-      return value.toTensor();
+      torch::Tensor t = value.toTensor();
+      if (t.scalar_type() == torch::kFloat16
+          || t.scalar_type() == torch::kFloat64)
+        return t.to(torch::kFloat32);
+      else
+        return t;
```

Comment:
I think checking type is redundant here, since it may be done internally in `.to(...)`. So this could be simplified into `value.toTensor().to(torch::kFloat32)`

Reply:
indeed this check is mandatory: if type is int[something] we do not want to convert (some models take int tensor as inputs, especially traced models)

Reply:
oh you're right :+1:

Reply:
I did not think about this until some ut fail ;)